contracts/gateway/OSTComposer.sol

```diff
@@ -0,0 +1,437 @@
+
+    /* Mapping of staker addresses to their StakerProxy. */
+    mapping (address => StakerProxy) public stakerProxies;
+
+    /* Stores number of all active stake request per staker. */
+    mapping(address => uint256) public activeStakeRequestCount;
+
+    /* Stores all the parameters of stake request based on stake request hash. */
+    mapping (bytes32 => StakeRequest) public stakeRequests;
+
+
+    /* Special Functions */
+
+    /**
+     * @notice Contract constructor.
+     *
+     * @param _organization Address of an organization contract.
+     */
+    constructor(
+        OrganizationInterface _organization
+    )
+        Organized(_organization)
+        public
+    {
+
+    }
+
+
+    /* External Functions */
+
+    /**
+     * @notice Staker calls the method to show its intention of staking. In order
+     *         to stake, the staked amount must first be transferred to
+     *         OSTComposer. Staker should approve OSTComposer for token
+     *         transfer.
+     *
+     * @param _gateway Address of the gateway to stake.
+     * @param _amount Amount that is to be staked.
+     * @param _beneficiary The address in the auxiliary chain where the utility
+     *                     tokens will be minted.
+     * @param _gasPrice Gas price that staker is ready to pay to get the stake
+     *                  and mint process done.
+     * @param _gasLimit Gas limit that staker is ready to pay.
+     * @param _nonce Staker nonce specific to gateway.
+     *
+     * @return stakeRequestHash_ A unique hash for stake request.
+     */
+    function requestStake(
+        EIP20GatewayInterface _gateway,
+        uint256 _amount,
+        address _beneficiary,
+        uint256 _gasPrice,
+        uint256 _gasLimit,
+        uint256 _nonce
+    )
+        external
+        mutex
+        returns (bytes32 stakeRequestHash_)
+    {
+        require(
+            _amount > uint256(0),
+            "Stake amount must not be zero."
+        );
+        require(
+            _beneficiary != address(0),
+            "Beneficiary address must not be zero."
+        );
+
+        require(
+            stakeRequestHashes[msg.sender][address(_gateway)] == bytes32(0),
+            "Request for this staker at this gateway is already in process."
+        );
+
+        stakeRequestHash_ = hashStakeRequest(
+            msg.sender,
+            address(_gateway),
+            _amount,
+            _beneficiary,
+            _gasPrice,
+            _gasLimit,
+            _nonce
+        );
+
+        stakeRequestHashes[msg.sender][address(_gateway)] = stakeRequestHash_;
+
+        StakerProxy stakerProxy = stakerProxies[msg.sender];
+
+        if(address(stakerProxy) == address(0)) {
+            stakerProxy = new StakerProxy(msg.sender);
+            stakerProxies[msg.sender] = stakerProxy;
+        }
+
+        require(
+            _gateway.getNonce(address(stakerProxy)) == _nonce,
```

Comment:
`Nonce` is being checked here as well as staker proxy. It seems like duplicate logic. 
cc/ @schemar